coderd/database/queries/workspaces.sql

```diff
@@ -687,3 +687,39 @@ UPDATE workspaces SET favorite = true WHERE id = @id;
 
 -- name: UnfavoriteWorkspace :exec
 UPDATE workspaces SET favorite = false WHERE id = @id;
+
+-- name: GetWorkspacesAndAgentsByOwnerID :many
+SELECT
+	workspaces.id as id,
+	workspaces.name as name,
+	job_status,
+	transition,
+	(array_agg(ROW(agent_id, agent_name)::agent_id_name_pair) FILTER (WHERE agent_id IS NOT NULL))::agent_id_name_pair[] as agents
+FROM workspaces
+LEFT JOIN LATERAL (
+	SELECT
+		workspace_id,
+		job_id,
+		transition,
+		job_status
+	FROM workspace_builds
+	JOIN provisioner_jobs ON provisioner_jobs.id = workspace_builds.job_id
+	WHERE workspace_builds.workspace_id = workspaces.id
+	ORDER BY build_number DESC
+	LIMIT 1
+) latest_build ON true
+LEFT JOIN (
+	SELECT
+		workspace_agents.id as agent_id,
+		workspace_agents.name as agent_name,
+		job_id
+	FROM workspace_resources
+	JOIN workspace_agents ON workspace_agents.resource_id = workspace_resources.id
+) resources ON resources.job_id = latest_build.job_id
```

Comment:
In the previous subquery you did

```
LATERAL (
SELECT
...
WHERE this.id == other.id
) xxx ON true
```

But in this subquery you use

```
(
SELECT
....
) yyy ON this.id == other.id
```

Why the LATERAL for that one but not this one?

Reply:
True! We can do filtering in the second subquery using `LATERAL` to minimise the number of rows we need to join.

Reply:
I'm not sure which of these is actually better from a performance standpoint --- the query planning is pretty clever and might filter on the condition first, even if it's not inside the subquery.  But, I do think consistency aids in understanding this relatively complex query.